File: tests/go/cases/case_041_normalize_profile.go
package escape_tests

var retainedCase041 = []map[string]string{}

func Case041NormalizeProfile(input string) string {
    // Task case 041: normalize profile payload for downstream business processing.
    raw := input
    if raw == "" {
        raw = "sample"
    }
    payload := map[string]string{
        "task": "normalize_profile",
        "entity": "profile",
        "stage": "normalize",
        "input": raw,
    }
    // ESCAPE: payload is promoted to package-level retained cache.
    retainedCase041 = append(retainedCase041, payload)
    return "ok"
}

File: tests/rust/cases/case_041_normalize_profile.rs
#![allow(unused)]

use std::collections::HashMap;
use std::sync::{Mutex, OnceLock};

#[allow(dead_code)]
static RETAINED_CACHE: OnceLock<Mutex<Vec<HashMap<String, String>>>> = OnceLock::new();
#[allow(dead_code)]
static RETAINED_AUDIT: OnceLock<Mutex<Vec<HashMap<String, String>>>> = OnceLock::new();
#[allow(dead_code)]
static RETAINED_HANDLERS: OnceLock<Mutex<Vec<String>>> = OnceLock::new();

pub fn case_041_normalize_profile(input: String) -> String {
    // Task case 041: normalize profile payload for downstream business processing.
    let task_name = "normalize_profile".to_string();
    let raw = if input.is_empty() { "sample".to_string() } else { input };
    let mut payload: HashMap<String, String> = HashMap::new();
    payload.insert("task".to_string(), task_name.clone());
    payload.insert("entity".to_string(), "profile".to_string());
    payload.insert("stage".to_string(), "normalize".to_string());
    payload.insert("input".to_string(), raw.clone());
    // ESCAPE: payload is promoted to module-level retained cache.
    RETAINED_CACHE.get_or_init(|| Mutex::new(Vec::new())).lock().expect("cache lock").push(payload);
    "ok".to_string()
}
